### PR TITLE
Made Stack Size parametrable (for #3760)

### PR DESCRIPTION
**Pull request by ptitSeb**
Made Stack Size parametrable.

By default, Stacks are 1MB. It is generaly a good default value, but it might be to small for some specific case, where recursion is (ab)used a lot, like for ticket #3760 .
In some other special case, with a lot of thread, smaller value might also be prefered.

### CODE REVIEW

File: lib/vm/src/trap/traphandlers.rs [hunk 1]
```diff
@@ -29,6 +29,18 @@ use wasmer_types::TrapCode;
 // On Arm64, the udf alows for a 16bits values, so we'll use the same 0xC? to store the trapinfo
 static MAGIC: u8 = 0xc0;
 
+static mut DEFAULT_STACK_SIZE: usize = 1024 * 1024;
```
Comment (Michael-F-Bryan) [inline lib/vm/src/trap/traphandlers.rs:32]: Do we need to use a `static mut` variable here? If we make it an `AtomicUsize` we can drop the `unsafe`.

**Review by Michael-F-Bryan — COMMENTED** [review]
I think it should be fine to make this a global variable here instead of trying to thread the stack size through `Tumables`. 

We should probably use an atomic or `Mutex` for the variable so we can drop the `unsafe`, though.

File: lib/vm/src/trap/traphandlers.rs (hunk 1 above)
Comment (ptitSeb) [reply lib/vm/src/trap/traphandlers.rs]: Yeah, it's much cleaner to use Atomic, and I don't think it will have much impact on performances...

**Comment by theduke** [comment]
I disagree, we very well may want different stack sizes per instance. 

I can think of a concrete use case where users can configure the stack size per workload...

This global atomic approach won't work for this. 

A global atomic won't work when multiple threads are constructing instances. 

In general I'm not a fan of a global static setting.

This should be threaded through, even though it's annoying. 

Note that tunables isn't ideal here either, because that would require constructing a separate engine.

@ptitSeb , does the chosen stack size affect compilation?

**Comment by theduke** [comment]
I'm not talking about changing the stack size for an existing instance, just configuring the size for a newly created one.

**Review by Michael-F-Bryan — APPROVED** [review]

**Pull request merged**